.github/workflows/docker-image-build-push.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and publish Docker image
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+
+jobs:
+  build_and_push_docker_image:
+    if: github.repository_owner == 'NethermindEth'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push dev
+        if: ${{ github.ref == 'refs/heads/develop' }}
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: |
+            nethermindeth/juno:${{github.run_number}}-dev
```

Comment:
what is `github.run_number` here exactly? commit sha?

Reply:
is not a commit SHA, it is a unique identifier for each run of a GitHub Actions workflow but rely on it we can easily find exact commit which triggered it:
![build-id](https://user-images.githubusercontent.com/16260675/222084017-8e499dd6-599e-4afd-a082-98c3c93c336d.PNG)

Reply:
Got it, feel free to merge when @IronGauntlets approves.